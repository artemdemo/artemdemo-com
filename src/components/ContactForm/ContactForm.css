.contact-form-loading {
  position: absolute;
}
  .contact-form-loading__loaded {
    display: none;
  }

.contact-form-iframe {
  visibility: hidden;
  width: 640px;
  height: 822px;
}
  .contact-form-iframe__loaded {
    visibility: visible
  }

.contact-form-icon {
  font-size: 150%;
  opacity: 0.4;
}
