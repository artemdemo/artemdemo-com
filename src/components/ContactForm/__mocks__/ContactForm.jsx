import React from 'react';

const ContactForm = props => (
    <div data-mock='ContactForm'>
        {props.children}
    </div>
);

export default ContactForm;