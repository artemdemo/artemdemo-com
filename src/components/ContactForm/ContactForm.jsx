import React from 'react';
import classnames from 'classnames';
import Icon from '../../components/Icon/Icon';
import './ContactForm.css';

const FORM_URL =
  'https://docs.google.com/forms/d/e/1FAIpQLSfbcq5MKljpEZg27bOOj5OqdBP4ARBdikUs4G2zuqIgBYE4KA/viewform?embedded=true';

class ContactForm extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      formLoaded: false,
    };
  }

  iframeLoaded() {
    this.setState({
      formLoaded: true,
    });
  }

  render() {
    return (
      <>
        <div
          className={classnames({
            'contact-form-loading': true,
            'contact-form-loading__loaded': this.state.formLoaded,
          })}
        >
          Form is loading...
          <span className="contact-form-icon">
            <Icon name="cog fa-spin fa-fw" />
          </span>
        </div>
        <iframe
          className={classnames({
            'contact-form-iframe': true,
            'contact-form-iframe__loaded': this.state.formLoaded,
          })}
          src={FORM_URL}
          onLoad={this.iframeLoaded.bind(this)}
          frameBorder="0"
          marginHeight="0"
          marginWidth="0"
        ></iframe>
      </>
    );
  }
}

export default ContactForm;
