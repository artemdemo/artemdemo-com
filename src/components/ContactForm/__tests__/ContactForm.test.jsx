import React from 'react';
import renderer from 'react-test-renderer';
import ContactForm from '../ContactForm';

jest.mock('../../Icon/Icon.jsx');

describe('ContactForm', () => {
    it('should render', () => {
        const tree = renderer.create(
            <ContactForm />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
