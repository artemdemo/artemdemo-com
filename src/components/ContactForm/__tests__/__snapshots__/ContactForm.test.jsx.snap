// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should render 1`] = `
Array [
  <div
    className="ContactForm__ContactFormLoading-r3mpp6-0 eaCgkI"
  >
    Form is loading...
    <span
      className="ContactForm__IconWrap-r3mpp6-2 bTFytk"
    >
      <div
        data-mock="Icon"
      >
        name: 
        cog fa-spin fa-fw
        title: 
        className: 
        inText: 
        false
      </div>
    </span>
  </div>,
  <iframe
    className="ContactForm__ContactFormIframe-r3mpp6-1 ihVECD"
    frameBorder="0"
    marginHeight="0"
    marginWidth="0"
    onLoad={[Function]}
    src="https://docs.google.com/forms/d/e/1FAIpQLSfbcq5MKljpEZg27bOOj5OqdBP4ARBdikUs4G2zuqIgBYE4KA/viewform?embedded=true"
  />,
]
`;
